crates/codegen/parser_templates/src/rust/cst_visitor.rs

```diff
@@ -0,0 +1,130 @@
+use std::{ops::Range, rc::Rc};
+
+use super::cst::*;
+use super::kinds::*;
+
+#[allow(unused_variables)]
+pub trait Visitor<E> {
+    fn enter_rule(
+        &mut self,
+        kind: RuleKind,
+        span: &Range<usize>,
```

Comment:
should we add `span` to all visitor methods? I would expect most use cases are able to just call `node.span()` when needed, rather than adding and ignoring (via `_`) an unused parameter everywhere.

Reply:
I prefer to have the final argument (the node itself) virtually always ignored. But this is an aesthetic judgement on which we can disagree :)

Reply:
On the other hand, if we are always passing all enum contents of `node`, in both `XXX_rule(kind, span, children)` and `XXX_token(kind, span, trivia)`, should we also pass a `node` parameter? I think the explicit contents would always be used, instead of always matching on the node to get the right variant, and using `unreachable!()` on the other arm.

Reply:
The node is just used in the rare case that you want to clone the `Rc` e.g. you are deriving a data structure such as an AST. Which is how I anticipate the AST being built BTW.

Reply:
And now  i recall my reasoning for not using the node in the common case is to avoid that pattern match in every method. The discrimination is already done in the visitor, as you point out.

Reply:
> The node is just used in the rare case that you want to clone the `Rc`

Good point. Let's keep as-is then.